packages/server/lib/modes/run.js

```diff
@@ -1019,14 +1025,25 @@ module.exports = {
       debug('waiting for socket to connect and browser to launch...')
 
       return Promise.join(
-        this.waitForSocketConnection(project, socketId)
-        .tap(() => {
-          debug('socket connected', { socketId })
-        }),
-        this.launchBrowser(options)
-        .tap(() => {
-          debug('browser launched')
-        }),
+        shouldLaunchBrowser
```

Comment:
Nit: might be more readable:

```js
if (shouldLaunchBrowser) {
  return Promise.join(...)
}

return Promise.join(...)
```

No need for the double ternary.

Reply:
There is a much leaner implementation indeed.